server/proxy/pf_channels.c

```diff
@@ -57,6 +64,8 @@ void pf_OnChannelConnectedEventHandler(void* data,
 	}
 	else if (strcmp(e->name, RDPGFX_DVC_CHANNEL_NAME) == 0)
 	{
+		pf_channels_wait_for_server_dynvc(ps);
```

Comment:
Why not just wait on server `drdynvc` when the client `drdynvc` event is triggered?

Reply:
Because client drdynvc event will be triggered even if original client did not connect with any dynamic channel. This is because proxy's client always opens DISP, because it has no idea whether if original client connected with disp or not.

Reply:
@akallabeth

Reply:
@kubistika sorry, did want to think a bit about it, but did not have time.
lets merge this one and (hopefully) come up with a better solution later.

Reply:
I'd really appreciate if you will have the time to help me figure another solution. I don't like it too but this is the only solution I found for this :/ (that makes sure channels will be opened)

Reply:
I'll keep an eye open for this, but currently a bit busy :/